openweathermap-ios/openweathermap-ios/Maps/ViewControllers/MapViewController.swift

```diff
@@ -123,3 +129,12 @@ extension MapViewController: CLLocationManagerDelegate {
         mapView.setRegion(region, animated: true)
     }
 }
+
+extension MapViewController: WeatherManagerDelegate {
+    func didUpdateWeather(_ weatherManager: WeatherManager, weather: WeatherModel) {
+        DispatchQueue.main.async {
```

Comment:
Why do you need dispatch on main if you are only setting properties?

Reply:
Because that part receives the data that will be updated interface. And according to apple's documentation, to avoid common errors resulting from updating the interface, broken data and crashes.

Reply:
I may be missing something, but I almost sure you are not performing any UI changes here to be necessary dispatching on the main thread. Your properties don't have any KVO attached either.